doc/languages-frameworks/perl.section.md

```diff
@@ -0,0 +1,179 @@
+3. In the fixup phase, it writes the propagated build inputs (`propagatedBuildInputs`) to the file `$out/nix-support/propagated-user-env-packages`. <command>nix-env</command> recursively installs all packages listed in this file when you install a package that has it. This ensures that a Perl package can find its dependencies.
+
+`buildPerlPackage` is built on top of ´stdenv´, so everything can be customised in the usual way. 
+For instance, the `BerkeleyDB` module has a ´preConfigure´ hook to generate a configuration file used by `Makefile.PL`:
+
+```nix
+{ buildPerlPackage, fetchurl, db }:
+
+buildPerlPackage rec {
+  name = "BerkeleyDB-0.36";
+
+  src = fetchurl {
+    url = "mirror://cpan/authors/id/P/PM/PMQS/${name}.tar.gz";
+    sha256 = "07xf50riarb60l1h6m2dqmql8q5dij619712fsgw7ach04d8g3z1";
+  };
+
+  preConfigure = ''
+    echo "LIB = ${db.out}/lib" > config.in
+    echo "INCLUDE = ${db.dev}/include" >> config.in
+  '';
+}
+```
+  
+Dependencies on other Perl packages can be specified in the ´buildInputs´ and ´propagatedBuildInputs´ attributes. 
+If something is exclusively a build-time dependency, use ´buildInputs´; if it’s (also) a runtime dependency, use ´propagatedBuildInputs´. 
+For instance, this builds a Perl module that has runtime dependencies on a bunch of other modules:
+
+```nix
+ClassC3Componentised = buildPerlPackage rec {
+  name = "Class-C3-Componentised-1.0004";
+  src = fetchurl {
+    url = "mirror://cpan/authors/id/A/AS/ASH/${name}.tar.gz";
+    sha256 = "0xql73jkcdbq4q9m0b0rnca6nrlvf5hyzy8is0crdk65bynvs8q1";
+  };
+  propagatedBuildInputs = [
+    ClassC3 ClassInspector TestException MROCompat
+  ];
+};
+```
+
+On Darwin, if a script has too many `-I<replaceable>dir</replaceable>` flags in its first line (its “shebang line”), it will not run. 
+This can be worked around by calling the `shortenPerlShebang` function from the `postInstall` phase:
+
+```nix
+{ stdenv, buildPerlPackage, fetchurl, shortenPerlShebang }:
+
+ImageExifTool = buildPerlPackage {
+  pname = "Image-ExifTool";
+  version = "11.50";
+
+  src = fetchurl {
+    url = "https://www.sno.phy.queensu.ca/~phil/exiftool/Image-ExifTool-11.50.tar.gz";
+    sha256 = "0d8v48y94z8maxkmw1rv7v9m0jg2dc8xbp581njb6yhr7abwqdv3";
+  };
+
+  buildInputs = stdenv.lib.optional stdenv.isDarwin shortenPerlShebang;
+  postInstall = stdenv.lib.optional stdenv.isDarwin ''
+    shortenPerlShebang $out/bin/exiftool
+  '';
+};
+```
+
+This will remove the `-I` flags from the shebang line, rewrite them in the `use lib` form, and put them on the next line instead. 
+This function can be given any number of Perl scripts as arguments; it will modify them in-place.
+
+## Generation from CPAN {#ssec-generation-from-CPAN}
```

Comment:
This section and the one below previously belonged under “Packaging Perl programs”. Is this change intentional?

Reply:
No, it was not intentional. When looking at the XML source it's difficult to keep track of this kind of detail. It's fixed by increasing the section level from '##' to '###'. Hope that's supported.